src/chapters/style/documentation.typ

```diff
@@ -0,0 +1,247 @@
+  ```
+]
+
+There are two kinds of documentation comments, inner and outer doc comments.
+Outer doc comments are placed right above the declaration they are attached to.
+
+#do-dont[
+  ```typst
+  /// Documentation for func
+  #let func() = { ... }
+
+  ```
+][
+  ```typst
+  /// Stray doc comment, not the doc for func
+
+  #let func() = {
+    /// Stray doc comment, not the doc for func
+    ...
+  }
+
+  /// Stray doc comment, not the doc for func
+  ```
+]
+
+Inner doc comments are used to document modules and must not have a declaration, instead they refer to the file they are placed in and may only be declared once and as the first non-comment item in a file.
+There may be regular comments before doc comments to allow placing internal documentation or license information at the top of the file.
+Inner doc comments are currently not permitted anywhere else.
+
+#do-dont[
+  ```typst
+  // optional leading comments
+
+  /// Module doc
+
+  /// Function or value doc
+  #let item = { ... }
+  ```
+][
+  ```typst
+  /// Function or value doc
+  #let item = { ... }
+
+  /// Stray doc comment, not the module or func doc
+  ```
+]
+
+Outer doc comments may be used on `let` bindings only.
+
+#do-dont[
+  ```typst
+  /// Function doc
+  #let func() = { ... }
+
+  /// Value doc
+  #let value = { ... }
+  ```
+][
+  ```typst
+  /// Stray doc comment, markup may not be documented
+  Hello World
+
+  /// Stray doc comment, imports my not be documented
+  #import "module.typ"
+
+  /// Stray doc comment, scopes may not be documented
+  #[
+    ...
+  ]
+  ```
+]
+
+Doc comments contain a description of the documented item itself, as well as an optional semantic trailer.
+The content of descriptions should generally be simple Typst markup and should not contain any scripting, (i.e. no loops, conditionals or function calls, except for `#link("...")[...]`), this allows the documentation to be turned into markdown or plain text for language servers to send to editors.
+
+== Description
+As mentioned before, the description should be simple, containing mostly markup and no scripting.
+
+== Semantic Trailer
+The semantic trailer is fully optional, starts with an empty doc comment line to separate it from the description, and may contain:
+- multiple `term` items for parameter type hints and descriptions,
+- a return type hint `-> type`,
+- and multiple property annotations (`#property("private")` or `#property("deprecated")`).
+
+Types in type lists or return type annotations may be separated by `|` to indicate that more than one type is accepted, the exact types allowed depend on the doc parser, but the built in types are generally supported.
+
+Parameter description and return types (if present) are placed tightly together, property annotations if present are separated using another empty doc comment line.
+
+Parameter documentation is created by writing a term item containing the parameter name, a mandatory type list in parentheses and optional description.
```

Comment:
How do we make description optional? I find that the description (content of a term item) is required by typst syntax.
```
/ a (int | float)
// error: expected colon
```

Reply:
Term descriptions can be left empty: `/ a (int | float):` would be valid. Although I thought I removed that part, I don't think one should document the types but add no description. I'll remove that.